pkg/nsm/apis/common/common.proto

```diff
@@ -28,17 +28,17 @@ message Empty{
 
 }
 
-message InterfaceParameters {
+message LocalMechanismParameters {
    map <string,string> interface_parameters = 1;
 }
 
-message Interface {
-   InterfaceType type = 1;
-   InterfacePreference preference = 2;
-   InterfaceParameters parmeters = 3;
+message LocalMechanism {
+   LocalMechanismType type = 1;
```

Comment:
How about to drop LocalMechanism here and have just Type, Preference and Parameters?

Reply:
I think we still need to distinguish LocalMechanismTypes from RemoteMechanismTypes.  That said, in 309 I do collapse down parameters somewhat.  I've also noticed that we don't seem to use Preference.  Since repeated preserves order, my thought was to at some point clip Preference and just rely on order to express Preference.

Reply:
@sbezverk Updated per IRC conversation.